Base64 video

MarkDaoust opened the pull request:
Embedding video was broken for python3, just giving a traceback.

It was loading the file's binary as bytes, then trying to ".encode" them to base64. Bytes don't have a ".encode" method. The "base64" module does this bytes-->bytes translation.

This should be python2 compatible. 

The decode transforms back to text form bytes for the ".format"

MarkDaoust committed:
fixed base64 embedding for video on python3
python 2 compatible

MarkDaoust committed:
Changed decode line: base64 can only produce ASCII characters

rgbkrk commented:
Thank you for the PR!

@sukisuki is one of the sprinters at PyCon and we weren't sure where to submit this patch (aim at the backports, forward port to `IPython/lib/display.py`.

Sorry if I advised the wrong direction, I now see that this patch also belongs in https://github.com/ipython/ipython/blob/master/IPython/core/display.py, which has not changed paths.

Should we submit a follow on PR?

minrk commented:
Typically we prefer to make and review PRs on master, then backport, rather than on the stable version and forward-port. I don't know how strict we want to be about that. For PRs that touch the split repos, they will have to be made as separate PRs against two repos, but since this is truly an IPython PR, our existing backport machinery will work correctly if it's made against master.

Carreau commented:
I would say that maybe for a couple of weeks (and especially pycon)
installing master is too painful, and we don't (yet) have docs.
But agree that we should relatively soon reask to PR on correct repo.

On Mon, Apr 13, 2015 at 12:22 PM, Min RK <notifications@github.com> wrote:

> Typically we prefer to make and review PRs on master, then backport,
> rather than on the stable version and forward-port. I don't know how strict
> we want to be about that. For PRs that touch the split repos, they will
> have to be made as separate PRs against two repos, but since this is truly
> an IPython PR, our existing backport machinery will work correctly if it's
> made against master.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ipython/ipython/pull/8312#issuecomment-92468581>.
>

rgbkrk commented:
@sukisuki headed out after we lost internet, but I did talk to him in person again. He had also written some tests and would be happy to target master. After we had gotten it to work in the notebook for him, it was clear we could come back to applying the changes again with master relying on tests for validity.

MarkDaoust committed:
added basic test
  - ensure video embedding doesn't outright fail
  - check that the generated url is correct

sukisuki commented:
Thanks for the input.
I'll retarget on master in a second.
